frontends/common/parser_options.cpp

```diff
@@ -454,19 +453,22 @@ void ParserOptions::closePreprocessedInput(FILE *inputStream) const {
 
 // From (folder, file.ext, suffix)  returns
 // folder/file-suffix.ext
-static cstring makeFileName(cstring folder, cstring name, cstring baseSuffix) {
-    Util::PathName filename(name);
-    Util::PathName newName(filename.getBasename() + baseSuffix + "." + filename.getExtension());
-    auto result = Util::PathName(folder).join(newName.toString());
-    return result.toString();
+static std::filesystem::path makeFileName(const std::filesystem::path &folder,
+                                          const std::filesystem::path &name,
+                                          std::string_view baseSuffix) {
+    std::filesystem::path newName(name.stem());
+    newName += baseSuffix;
+    newName += name.extension();
+
+    return folder / newName;
 }
 
 bool ParserOptions::isv1() const { return langVersion == ParserOptions::FrontendVersion::P4_14; }
 
 void ParserOptions::dumpPass(const char *manager, unsigned seq, const char *pass,
                              const IR::Node *node) const {
     if (strncmp(pass, "P4::", 4) == 0) pass += 4;
-    cstring name = cstring(manager) + "_" + Util::toString(seq) + "_" + pass;
+    std::string name = absl::StrCat(manager, "_", seq, "_", pass);
```

Comment:
It's a shame we can't just use `+` instead of having to invoke `StrCat`. But I know we can't with the current types of the elements being concatenated.

Reply:
Yes. Some stdlib implementations even return proxy object as a result operator+ to make things a bit more efficient. But it's not the case here. Anyway, the past code created lots of cstrings for no particular case...